climakitae/core/data_interface.py

```diff
@@ -1577,3 +1586,426 @@ def _display_select(self):
+        # Use the catalog to find the valid values in the list
+        valid_options = np.unique(cat_df[key].values)
+        if (
+            val == None
+        ):  # If the user didn't input anything for that key, set the values to all the valid options
+            d[key] = valid_options
+            continue  # Don't finish the loop
+        # If the input value is not in the valid options, see if you can help the user out
+        elif val not in valid_options:
+            # Perhaps they didn't capitalize it correctly?
+            if val.lower().capitalize() in valid_options:
+                d[key] = [val.lower().capitalize()]
+                continue  # Don't finish the loop
+
+            # This catches any common bad inputs for resolution: i.e. "3KM" or "3km" instead of "3 km"
+            if key == "resolution":
+                try:
+                    good_resolution_input = val.lower().split("km")[0] + " km"
+                    if good_resolution_input in valid_options:
+                        d[key] = [good_resolution_input]
+                        continue
+                except:
+                    pass
+
+            print("Input " + key + "='" + val + "' is not a valid option.")
+            # Use difflib package to make a guess for what the input might have been
+            # For example, if they input "statistikal" instead of "Statistical", difflib will find "Statistical"
+            # Change the cutoff to increase/decrease the flexibility of the function
+            closest_option = difflib.get_close_matches(val, valid_options, cutoff=0.59)
+            if len(closest_option) == 0:
+                # Sad! No closest options found. Just set the key to all valid options
+                print("Valid options: " + ", ".join(valid_options))
+                raise ValueError("Bad input")
+            else:  # difflib made a guess! Set the key to the guess (or, guesses)
+                if len(closest_option) == 1:
+                    print("Did you mean '" + closest_option[0] + "'?")
+                else:
+                    print(
+                        "Did you mean: \n- "
+                        + "\n- ".join([x + "?" for x in closest_option])
+                    )
+                d[key] = closest_option
+    # Dictionary values should be a list for the pandas logic to work correctly
+    for key, val in zip(d.keys(), d.values()):
+        if type(val) not in (list, np.ndarray):
+            d[key] = [val]
+
+    # Subset the catalog with the user's inputs
+    cat_subset = cat_df[
+        (cat_df["variable"].isin(d["variable"]))
+        & (cat_df["downscaling_method"].isin(d["downscaling_method"]))
+        & (cat_df["resolution"].isin(d["resolution"]))
+        & (cat_df["timescale"].isin(d["timescale"]))
+        & (cat_df["scenario"].isin(d["scenario"]))
+    ].reset_index(drop=True)
+    if len(cat_subset) == 0:
+        print("No data found for your input values")
```

Comment:
A nice to have is to point the users to a place where they may have a better chance. Perhaps the data catalog here: https://analytics.cal-adapt.org/data/catalog/